test/test_alert_manager.cpp

```diff
@@ -223,6 +255,42 @@ TORRENT_TEST(extensions)
 	TEST_EQUAL(plugin_alerts[0], 70);
 	TEST_EQUAL(plugin_alerts[1], 70);
 	TEST_EQUAL(plugin_alerts[2], 70);
+
+	for (int i = 0; i < 35; ++i)
+		mgr.emplace_alert<torrent_finished_alert>(torrent_handle());
```

Comment:
I think it would make sense to split this up in multiple test cases. Right now there are just a bunch of different aspects tested  in one sequence, making it harder to run them in isolation.